client/src/app/containers/Game/useGame.ts

```diff
@@ -0,0 +1,235 @@
+import axios from 'axios';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useAuth } from '../../../auth/useAuth';
+import { RootState } from '../../../store/store';
+import { useAlert } from '../Alert/useAlert';
+import { Card, Game, Phase } from './models';
+import {
+    actions,
+    CountCribAction,
+    CountHandAction,
+    CribAction,
+    CutAction,
+    DealAction,
+    GameAction,
+    PegAction,
+} from './slice';
+import { CreateGameResponse } from './types';
+
+interface Result {
+    game: Game;
+    selectedCards: Card[];
+    createGame: (playerIDs: string[]) => Promise<void>;
+    getGame: (id: number) => Promise<void>;
+    refreshGame: () => Promise<void>;
+    toggleSelectedCard: (c: Card) => void;
+    clearSelectedCards: () => void;
+    submitDealAction: (a: DealAction) => Promise<void>;
+    submitBuildCribAction: (a: CribAction) => Promise<void>;
+    submitCutDeckAction: (a: CutAction) => Promise<void>;
+    submitPegAction: (a: PegAction) => Promise<void>;
+    submitCountHandAction: (a: CountHandAction) => Promise<void>;
+    submitCountCribAction: (a: CountCribAction) => Promise<void>;
+}
+// getPlayerAction returns the JSON struct which the server knows
+// how to interpret
+interface ServerCard {
+    s: number;
+    v: number;
+}
+interface ActionRequest {
+    pID: string;
+    gID: number;
+    o: ReturnType<typeof mapPhaseToOverComes>;
```

Comment:
:mag: :mage: Is this some witchcraft?

Reply:
TypeScript can figure out what the function `mapPhaseToOverComes` returns. That's what this is.

Reply:
Dope!